server/applicationset/applicationset.go

```diff
@@ -208,11 +212,20 @@ func (s *Server) Create(ctx context.Context, q *applicationset.ApplicationSetCre
 		if err != nil {
 			return nil, fmt.Errorf("unable to generate Applications of ApplicationSet: %w", err)
 		}
-
 		statusMap := appsetstatus.GetResourceStatusMap(appset)
-		statusMap = appsetstatus.BuildResourceStatus(statusMap, apps)
+		// if any application uses apptree instead of inline, grab all the statuses from the cache instead of the CRs
+		isAppTree := appsetstatus.AppsUseAppTree(apps)
```

Comment:
I guess I could expose the config from `argocd-cm-params` to the server and pass it all the way down but we are already doing a check like this for applications in the [server](https://github.com/argoproj/argo-cd/blob/1645d576fd0c8e9e822037d5ff6dd4469aeaf0ff/server/application/application.go#L2650C71-L2650C78).